addons/common/functions/fnc_registerAmmoFiredEvent.sqf

```diff
@@ -0,0 +1,54 @@
+/*
+ * Author: esteldunedain
+ * Register an ammo fired event that should be launched when ammo following a
+ * given criteria is fired. The criteria is analyzed on mission start and is
+ * cached.
+ *
+ * Arguments:
+ * 0: Name of the event <STRING>
+ * 1: Condition code <CODE>
+ * 2: Call for the local player on foot? <BOOL>
+ * 3: Call for non local players on foot? <BOOL>
+ * 4: Call for non players on foot? <BOOL>
+ * 5: Call for the local player vehicle? <BOOL>
+ * 6: Call for non local players vehicles? <BOOL>
+ * 7: Call for non players vehicles? <BOOL>
```

Comment:
> What is a non-local player vehicle? What if the player is the commander?

Actually what matters for remote vehicles is if a player is manning the turret or not. If the vehicle is in "manual fire" actually effective commander is checked, etc.

The names are a bit ambiguous, I know. But the underlying functionality is tried and true.

I think replacing "player on foot" by "player soldier" or "player unit", etc should clear some of the confusion.

Don't forget that the purpose of all these is that most things we do for shots we only do for players. But sometimes each player handles their own stuff only; and sometimes each client handles all players locally.